mesonbuild/dependencies/mpi.py

```diff
@@ -244,3 +247,54 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
         self.compile_args = ['-I' + incdir, '-I' + os.path.join(incdir, post)]
         if self.language == 'fortran':
             self.link_args.append('-l' + os.path.join(libdir, 'msmpifec'))
+
+
+class IMPIDependency(SystemDependency):
+
+    """Intel(R) MPI for Windows."""
+
+    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
```

Comment:
Nit: don't quote this.

```suggestion
    def __init__(self, name: str, env: Environment, kwargs: T.Dict[str, T.Any],
```

It is redundant due to https://peps.python.org/pep-0563/ use of `from __future__ import annotations`.

Reply:
I know it is redundant, I was just following the rules of pre-existent code. I'l fix.

Reply:
@eli-schwartz Sorry, while your comment is right, you are missing the full context, look at the top of the file.:

```python
...
if T.TYPE_CHECKING:
    ...
    from ..environment import Environment
    ...
```

Reply:
Not certain what you mean by that? The whole point of `from __future__ import annotations` is that it has the effect of implicit quotes around ALL type annotations in the file.

(Pre-existing code simply predated the use of the `__future__` import. For new code, we should avoid outdated hacks. Rewriting old code is not necessary unless those lines are already being edited for some other reason.)

Reply:
Oh, you are right, sorry for the noise.